stor/obs.py

```diff
@@ -128,7 +128,7 @@ def normpath(self):
         normed = posixpath.normpath('/' + str(self)[len(self.drive):])[1:]
         return self.path_class(self.drive + normed)
 
-    def read_object(self):
+    def read_object(self, mode='r'):
```

Comment:
If the intention is to use `with stor.Path('dx://path/to/file).open(mode='rb')` to access dx files, this is not really necessary because mode is already set in the `self` object by this point. You can remove it from `dx.py`, `s3.py` and `swift.py` as well. You'd just do in `dx.py`:
```
def read_object(self):
    ....
    file_handler = dxpy.DXFile(dxid=self.canonical_resource, project=self.canonical_project, mode=self.mode)
    ....
    if self.mode == 'r':
        ....
```

If there was some other reasoning behind this, you can disregard my comment.

Reply:
Thanks for the feedback. I just realized that `OBSFile` defined `mode` i was looking for it in `OBSPath` instead and not finding it.

I'll make the change but I'm not really sure about when when to use `open()` vs `read_object()`